File: swagger/models/get_client_contracts_request.py
from .base_model import BaseModel


class GetClientContractsRequest(BaseModel):
	"""
	no description available

	Attributes:

		client_id
		The ID of the client.

		cross_regional_lookup
		When `true`, indicates that the requesting client’s cross regional
		contracts are returned, if any.<br />  When `false`, indicates that
		cross regional contracts are not returned.

		client_associated_sites_offset
		Determines how many sites are skipped over when retrieving a
		client’s cross regional contracts. Used when a client ID is linked
		to more than ten sites in an organization. Only a maximum of ten site
		databases are queried when this call is made and `CrossRegionalLookup`
		is set to `true`. To change which sites are queried, change this
		offset value.  Default: **0**

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'ClientId': 'client_id',
		'CrossRegionalLookup': 'cross_regional_lookup',
		'ClientAssociatedSitesOffset': 'client_associated_sites_offset',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'client_id': 'ClientId',
		'cross_regional_lookup': 'CrossRegionalLookup',
		'client_associated_sites_offset': 'ClientAssociatedSitesOffset',
		'limit': 'Limit',
		'offset': 'Offset',
		}

	client_id = None
	cross_regional_lookup = None
	client_associated_sites_offset = None
	limit = None
	offset = None



	def __init__(self, client_id):
		self.client_id = client_id

File: swagger/Models/GetClientContractsRequest.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class GetClientContractsRequest extends BaseModel
{
	/**
	 * The ID of the client.
	 * 
	 * @var string 
	 */
	public $clientId;
	/**
	 * When `true`, indicates that the requesting client’s cross regional
	 * contracts are returned, if any.<br />  When `false`, indicates that
	 * cross regional contracts are not returned.
	 * 
	 * @var boolean 
	 */
	public $crossRegionalLookup;
	/**
	 * Determines how many sites are skipped over when retrieving a
	 * client’s cross regional contracts. Used when a client ID is linked
	 * to more than ten sites in an organization. Only a maximum of ten site
	 * databases are queried when this call is made and `CrossRegionalLookup`
	 * is set to `true`. To change which sites are queried, change this
	 * offset value.  Default: **0**
	 * 
	 * @var integer format:int32
	 */
	public $clientAssociatedSitesOffset;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer format:int32
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer format:int32
	 */
	public $offset;
	/**
	 * Constructor
	 */
	public function __construct($clientId)
	{
		$this->clientId = $clientId;
	}

	protected function getInputMap()
	{
		return [
			'ClientId' => ['clientId', 'string', null],
			'CrossRegionalLookup' => ['crossRegionalLookup', 'boolean', null],
			'ClientAssociatedSitesOffset' => ['clientAssociatedSitesOffset', 'integer', 'int32'],
			'Limit' => ['limit', 'integer', 'int32'],
			'Offset' => ['offset', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'clientId' => ['ClientId', 'string', null],
			'crossRegionalLookup' => ['CrossRegionalLookup', 'boolean', null],
			'clientAssociatedSitesOffset' => ['ClientAssociatedSitesOffset', 'integer', 'int32'],
			'limit' => ['Limit', 'integer', 'int32'],
			'offset' => ['Offset', 'integer', 'int32'],
			];
	}
}
